crml/staking/src/lib.rs

```diff
@@ -999,21 +1006,28 @@ decl_module! {
 				Error::<T>::NoMoreChunks,
 			);
 
-			let mut value = value.min(ledger.active);
-
-			if !value.is_zero() {
-				ledger.active -= value;
-
-				// Avoid there being a dust balance left in the staking system.
-				if ledger.active < T::Currency::minimum_balance() {
-					value += ledger.active;
-					ledger.active = Zero::zero();
-				}
+			if ledger.active.is_zero() || value.is_zero() {
+				return Ok(());
+			}
 
-				let era = Self::current_era() + T::BondingDuration::get();
+			// If active stake drops below the minimum bond threshold or currency minimum
+			// then the entirety of the stash should be scheduled to unlock.
+			// Care must be taken to ensure that funds are still at stake until the unlocking period is over.
+			let remaining_active = ledger.active.checked_sub(&value).unwrap_or(Zero::zero());
+			let era = Self::current_era() + T::BondingDuration::get();
+			if remaining_active < T::Currency::minimum_balance() || remaining_active < Self::minimum_bond() {
```

Comment:
I think we should only check the active has not gone under the minimum bond. The currency's minimum balance doesn't seem to be relevant in the context of bonding.